src/test/scala/org/allenai/pipeline/TestProducer.scala

```diff
@@ -67,6 +67,22 @@ class TestProducer extends UnitSpec with BeforeAndAfterAll {
     otherStep.get should equal(pStep.get)
   }
 
+  "PersistedProducer" should "always read from file if caching disabled" in {
```

Comment:
The use case is in Spark, so you can disable caching to force reading of an `RDD` from disk instead of recomputing it.